File: backend/src/main/java/ipeps/pwd/wallet/payload/createPayload/DocumentCreatePayload.java
package ipeps.pwd.wallet.payload.createPayload;

import ipeps.pwd.wallet.entity.Employee;
import ipeps.pwd.wallet.entity.Organization;
import ipeps.pwd.wallet.entity.Transaction;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DocumentCreatePayload {
    private String name;
    private String description;
    private String path;
    private String type;
    private boolean freeAccess;
    private Transaction transaction; //string transaction_id
    private Employee employee;//string employee_id
    private Organization organization;
}

File: frontend/src/app/modules/document/Model/payload/document-create.payload.ts
import { PayloadInterface } from '@shared/model';
import { EmployeeDto } from '@employee/model';
import { OrganizationDto } from '@organization/model';
import { TransactionDto } from '../../../wallet/model/dto/transaction.dto';

export interface DocumentCreatePayload extends PayloadInterface {
  name: string;
  description: string;
  path: string;
  type: string;
  freeAccess: boolean;
  employee: EmployeeDto;
  organization: OrganizationDto;
  transaction: TransactionDto;
}
